base/src/types/mod.rs

```diff
@@ -240,6 +242,21 @@ pub struct TypeVariable {
 #[cfg_attr(feature = "serde_derive", serde(de_parameters = "T"))]
 #[cfg_attr(feature = "serde_derive", serde(serialize_state = "SeSeed"))]
 #[cfg_attr(feature = "serde_derive", serde(bound(serialize = "Id: SerializeState<SeSeed>")))]
+pub struct Skolem<Id> {
+    #[cfg_attr(feature = "serde_derive", serde(state))] pub name: Id,
+    pub id: u32,
+    #[cfg_attr(feature = "serde_derive", serde(state))] pub kind: ArcKind,
+}
```

Comment:
Might be worth adding some doc comments to each of these types.

Reply:
Yep, going to newtype/alias `u32` as well since it has started leaking out in a few to many places.